sql/schemachanger: Log events for CREATE/DROP POLICY

spilchen opened the pull request:
Previously, RLS policy DDL actions were not logged in system.eventlog, making it harder to track changes during investigations. This change ensures that CREATE POLICY and DROP POLICY operations are properly logged.

Epic: CRDB-45203
Release note: none

Informs #136696

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/140970)
<!-- Reviewable:end -->

spilchen committed:
sql/schemachanger: Log events for CREATE/DROP POLICY

Previously, RLS policy DDL actions were not logged in system.eventlog,
making it harder to track changes during investigations. This change
ensures that CREATE POLICY and DROP POLICY operations are properly
logged.

Epic: CRDB-45203
Release note: none

Informs #136696

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 10 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/140970#-:-OIkE4LzBjoCeqgcFpUd:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi, @Abhinav1299, @arjunmahishi, and @xinhaoz)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+
